iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/checkpoint/CheckpointManager.java

```diff
@@ -345,12 +344,13 @@ public long getFirstValidWALVersionId() {
   }
 
   /** Update wal disk cost of active memTables. */
-  public void updateCostOfActiveMemTables(Map<Long, Long> memTableId2WalDiskUsage) {
+  public void updateCostOfActiveMemTables(
+      Map<Long, Long> memTableId2WalDiskUsage, double compressionRate) {
     for (Map.Entry<Long, Long> memTableWalUsage : memTableId2WalDiskUsage.entrySet()) {
       memTableId2Info.computeIfPresent(
           memTableWalUsage.getKey(),
           (k, v) -> {
-            v.addWalDiskUsage(memTableWalUsage.getValue());
+            v.addWalDiskUsage((long) (memTableWalUsage.getValue() * compressionRate));
```

Comment:
It's not about the file size. Multiple memtables will share one wal file, so we cannot get the disk usage of one memtable by getting the size of wal file directly.

Reply:
I never mentioned the file size.
I assume "memTableWalUsage.getValue() * compressionRate" is the compressed size, or the actual IO amount.
However,  this calculation may result in some precision loss ("b * (a/b)" is very likely not "b" in computers), and you can directly pass down the compressed size and use it. Why not use the compressed size directly?

Reply:
First, in this scene, we **do not** need a very specific value to control it. Both methods will work, but this way is simple(in case of editting code).
Second, it's difficult to calculate the extractly I/O size of each memtable, because the every WALBuffer may contain WAL Entry from multiple memtables. After compression, we could not get the extractly size of each memtable occupied.